### PR TITLE
Add test for function name being assigned in function

### PR DESCRIPTION
**Pull request by ranjanan**
This adds a test for #4914.

### CODE REVIEW

File: test/functional.jl [hunk 1]
```diff
@@ -513,3 +513,11 @@ let s = "Monkey 🙈🙊🙊"
     @test tf(2) == "Mo|nk|ey| 🙈|🙊🙊"
     @test tf(1) == "M|o|n|k|e|y| |🙈|🙊|🙊"
 end
+
+# Issue 4914
+let
```
Comment (tkelman) [inline test/functional.jl]: core is probably better for this. functional is for higher order functional programming

Comment (ranjanan) [inline test/functional.jl]: Right, I'll shift.

**Comment by ranjanan** [comment]
@tkelman is there anything I need to add to this PR?

**Comment by ranjanan** [comment]
@vtjnash Could you please help review this code and let me know if I should change anything?

File: test/core.jl [hunk 2]
```diff
@@ -4461,3 +4461,11 @@ end
 #end
 #@test local_innersig(Int32(2)) == ((Int32(2), Int32(1), Int32(2)im), (Int32(2), UInt32(1)))
 #@test local_innersig(Int64(3)) == ((Int64(3), Int64(1), Int64(3)im), (Int64(3), UInt64(1)))
+
+# Issue 4914
+let
+    j(j) = j
+    @test j(1) == 1
+    k(x) = (k = x; x)
```
Comment (JeffBezanson) [inline test/core.jl]: The example in the issue is `(k = x; k)`.

**Comment by ranjanan** [comment]
Looks like a LibGit2 error on Appveyor and a Travis timeout

**Pull request merged**